### PR TITLE
Add oauth_body_hash

### PR DESCRIPTION
**Pull request by webjunkie**
Add body_hash capability to server as in
http://oauth.googlecode.com/svn/spec/ext/body_hash/1.0/oauth-bodyhash.html

### CODE REVIEW

**Comment by joestump** [comment]
@webjunkie thanks for the PR! @jaitaiwan can you take a look?

File: oauth2/__init__.py [hunk 1]
```diff
@@ -695,8 +707,9 @@ class Server(object):
     version = OAUTH_VERSION
     signature_methods = None
 
-    def __init__(self, signature_methods=None):
+    def __init__(self, signature_methods=None, body_hashing=False):
```
Comment (joestump) [inline oauth2/__init__.py]: Can we make this `check_body_hashing`? Or `verify_body_hash`?

**Comment by jaitaiwan** [comment]
Is this the same issue as #138? If so, the body should always be hashed even if empty according to spec...

**Comment by joestump** [comment]
@jaitaiwan that's true, but I don't think body hashing itself is part of the OAuth specification. In other words, if you implement it you need to do this, but the providers don't have to implement body hashing. At least that's how I've understood this PR.

**Pull request closed without merging**